paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -444,11 +509,8 @@ internal class PaymentSheetViewModel @Inject internal constructor(
 
     @Suppress("ComplexCondition")
     private fun paymentSelectionWithCvcIfEnabled(paymentSelection: PaymentSelection?): PaymentSelection? {
-        return if (
-            (isCvcRecollectionEnabled() || isCvcRecollectionEnabledForDeferred()) &&
-            paymentSelection is PaymentSelection.Saved &&
-            paymentSelection.paymentMethod.type == PaymentMethod.Type.Card
-        ) {
+        if (paymentSelection !is PaymentSelection.Saved) return paymentSelection
```

Comment:
Not necessary for this PR, but we should pull this logic out into a common area and unit test it.